concepts/default-enarx-trust-process.md

```diff
@@ -0,0 +1,480 @@
+# TBD: Default Enarx trust process
+- Authors: [Mike Bursell](mike@p2ptrust.org)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-03-17
+- Status Note: under discussion
+- Start Date: 2020-03-17 (date you started working on this idea)
```

Comment:
> I think some diagrams would help reduce the amount of prose here. The state machine could be represented as a diagram and then each state could contain prose to describe it.

I think you're right, but I've tried to stick with standard RFC plain text.  I can _try_ to do state machines in ASCII art, I suppose: what do you think?  Alternatively, some pictures could be referenced.

> Furthermore, the process flow description could be factored out into something like an mscgen process flow diagram which would further reduce the amount of prose needed. In many cases we won't eliminate prose here, but I think it would allow for more concision.

Again, text is good and unambiguous, but... pictures are more concise.

> I'm also hesitant to approve an RFC with TODOs still embedded in the text.

Agreed.  I think that all(?) of the TODOs refer to other RFCs awaiting approval.  Any others should probably be issues.

Reply:
For the state machines I was thinking of dot graphs. For example: [source](https://github.com/enarx/enarx/blob/master/docs/attestation/amd/sev/certchain.dot), [output](https://github.com/enarx/enarx/blob/master/docs/attestation/amd/sev/certchain.dot.png).

For the process flow diagrams I was thinking of mscgen graphs. For example: [source](https://github.com/enarx/enarx/blob/master/docs/attestation/amd/sev/process.msc), [output](https://github.com/enarx/enarx/blob/master/docs/attestation/amd/sev/process.msc.png)

The example process flow diagram above is quite complex, so I think it misrepresents the learning curve to someone who is new to its syntax (like me)... but the graphs for this RFC can be quite basic, I think the visual representation of the process flow or the state machine would just pay back dividends in terms of making sense of it all.